addons/l10n_vn_viettel/models/account_move.py

```diff
@@ -0,0 +1,398 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+import logging
+_logger = logging.getLogger(__name__)
+import urllib3
+try:
+    import certifi
+except ImportError:
+    certifi = None
+    _logger.warning(_('Could not import certifi. Please see https://pypi.org/project/certifi/. SSL verification is disabled for communications with Viettel.'))
+from odoo import tools
+
+
+inv_fields_mapping = {
+        "invoiceIssuedDate": "create_date",
+        "invoiceNote": "summary",
+        "buyerName": "customer_name",
+        "buyerLegalName": "customer_legal_name",
+        "buyerTaxCode": "customer_tax_code",
+        "buyerAddressLine": "customer_address",
+        "buyerEmail": "customer_email",
+        "transactionUuid": "transaction_uuid",
+        # seller info
+        "sellerLegalName": "company_name",
+        "sellerTaxCode": "company_tax_code",
+        "sellerAddressLine": "company_address",
+        "sellerPhoneNumber": "company_number",
+        "sellerWebsite": "company_website",
+        "sellerBankAccount": "company_bank_account",
+        "sellerBankName": "company_bank_name",
+
+        "paymentMethodName": "payment_method_name",
+        # summary
+        "sumOfTotalLineAmountWithoutTax": "total_line_amount_wo_tax",
+        "totalAmountWithoutTax": "amount_untaxed",
+        "totalTaxAmount": "amount_tax",
+        "totalAmountWithTax": "amount_total",
+        "totalAmountWithTaxInWords": "amount_total_text",
+        "taxPercentage": "total_tax_percent",
+        "taxBreakdowns": "tax_list",
+}
+
+inv_line_fields_mapping = {
+    "lineNumber": "sequence",
+    "itemCode": "product_ref",
+    "itemName": "product_name",
+    "unitName": "uom_name",
+    "unitPrice": "price_unit",
+    "quantity": "quantity",
+    "itemTotalAmountWithoutTax": "price_subtotal",
+    "taxAmount": "amount_tax",
+    "taxableAmount": "taxable_amount",
+    "taxPercentage": "tax_percent",
+}
+
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    payment_journal_id = fields.Many2one('account.journal', domain="[('type', 'in', ('bank', 'cash')), ('bank_account_id', '!=', False)]", string="Payment Journal", states={'open': [('readonly', True)]})
+    l10n_vn_transaction_id = fields.Char(copy=False)
+    l10n_vn_invoice_no = fields.Char(string='Invoice No (Số hóa đơn)', copy=False)
+    l10n_vn_reservation_code = fields.Char(string='Reservation Code (Mã số bí mật)', copy=False)
+    l10n_vn_uuid = fields.Char(readonly=True, copy=False)
+    l10n_vn_is_send_mail_success = fields.Boolean(default=False, readonly=True, copy=False)
+    l10n_vn_transaction_date = fields.Char(string='Transaction ID (Ngày lập)', readonly=True, copy=False)
+    l10n_vn_file = fields.Binary(readonly=True, copy=False)
+    l10n_vn_vas_invoice_date = fields.Datetime(compute='_compute_l10n_vn_vas_invoice_date', store=True)
+    l10n_vn_tax_invoice_number = fields.Char(copy=False)
```

Comment:
Need some help with the explanation about what the date represents, why it is different than the other dates, ...

Reply:
It's the date of the registration on the platform, which is not the date of the invoice